models/issues/issue.go

```diff
@@ -92,6 +92,15 @@ func (err ErrIssueWasClosed) Error() string {
 	return fmt.Sprintf("Issue [%d] %d was already closed", err.ID, err.Index)
 }
 
+type IssueClosedStatus int8
+
+const (
+	IssueClosedStatusCommon   IssueClosedStatus = iota // 0 close issue without any state.
+	IssueClosedStatusArchived                          // 1
+	IssueClosedStatusResolved                          // 2
+	IssueClosedStatusStale                             // 3
+)
```

Comment:
As per discussion in https://github.com/go-gitea/gitea/issues/22793, I would request only two states: `completed` and `notPlanned.`